nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -952,7 +952,47 @@ VBufStorage_fieldNode_t* GeckoVBufBackend_t::fillVBuf(
 			// The object has no text, but we do want to render its children.
 			auto [varChildren, accChildRes] = getAccessibleChildren(pacc, 0, childCount);
 			if (S_OK != accChildRes || varChildren.size() == 0) {
-				LOG_ERROR(L"AccessibleChildren failed (count: " << childCount << L"), res: " << accChildRes);
+				std::wstringstream msg;
+				msg << L"AccessibleChildren failed (count: " << childCount << L"), res: " << accChildRes;
+				switch (accChildRes) {
+				case E_NOINTERFACE:
+					msg << L" (E_NOINTERFACE, No such interface supported)";
+					// This is certainly an error, and indicates a bug in the IA2 provider.
+					LOG_ERROR(msg.str());
+					break;
+				case RPC_E_DISCONNECTED:
+					msg << L" (RPC_E_DISCONNECTED, object invoked has disconnected from its clients.)";
+					// RPC_E_DISCONNECTED indicates that the parent died since the query
+					// to accChildCount, and returning no children indicates that all children
+					// died since the call to accChildCount.
+					// Even if the children had been rendered, they were removed immediately thereafter.
+					// Therefore log at debug level, this is expected to occur in dynamic content.
+					LOG_DEBUG(msg.str());
+					break;
+				case CO_E_OBJNOTCONNECTED:
+					msg << L" (CO_E_OBJNOTCONNECTED, Object is not connected to server)";
+					LOG_DEBUG(msg.str());
+					break;
+				case S_FALSE:
+					if (varChildren.size() == 0 && childCount > 0) {
+						msg << L" (S_FALSE, expected childcount, got no children, children may have been removed from document.)";
+						// Returning no children indicates that all children
+						// died since the call to accChildCount.
+						// Even if the children had been rendered, they were removed immediately thereafter.
+						// Therefore log at debug level, this is expected to occur in dynamic content.
+						LOG_DEBUG(msg.str());
+						break;
+					}
+					else {
+						// theoretically this cause can not be hit, included only for the benefit of developers

```

Comment:
On the contrary, it can be hit if, for example:

1. accChildCount returned 2.
2. One child died.
3. AccessibleChildren will return 1 child and S_FALSE (because 2 were requested but only 1 was returned).

Reply:
Looking at this again, I don't think we should try to distinguish between
- `S_FALSE` and got no children 
- `S_FALSE` and got less than `childCount`

I'm going to merge both `S_FALSE` handlers together.